pkg/controller/namespace/namespace_controller_utils.go

```diff
@@ -60,11 +61,22 @@ type operationKey struct {
 
 // operationNotSupportedCache is a simple cache to remember if an operation is not supported for a resource.
 // if the operationKey maps to true, it means the operation is not supported.
-type operationNotSupportedCache map[operationKey]bool
+type operationNotSupportedCache struct {
+	lock sync.Mutex
+	m    map[operationKey]bool
+}
 
 // isSupported returns true if the operation is supported
 func (o operationNotSupportedCache) isSupported(key operationKey) bool {
-	return !o[key]
+	o.lock.Lock()
+	defer o.lock.Unlock()
+	return !o.m[key]
```

Comment:
use a RWLock, the first namespace deleted will populate this map and we'll never write to it again